### PR TITLE
Add support for neverBlock on AsyncAppenders

### LINKED ISSUE #1916: Expose neverBlock option for async appender
logback offers an option, `neverBlock`, to make it so that an async appender will drop events rather than block when the queue fills. The default for this option is `false`, meaning the async appender will, in fact, block. I'd like to guard against this scenario in my application and prevent threads from backing up waiting on the appender by setting the option to `true`.

I'd propose Dropwizard config expose this option in the configuration block.

### PR DESCRIPTION
**Pull request by markelliot**
fixes #1916

### CODE REVIEW

File: docs/source/manual/core.rst [hunk 1]
```diff
@@ -924,6 +924,25 @@ single ``Logger``:
 
 .. _man-core-testing-applications:
 
+Common Logging Configuration
```
Comment (markelliot) [inline docs/source/manual/core.rst]: not really sure where to put common settings in docs, LMK if you'd like me to move this somewhere else.

Comment (nickbabcock) [inline docs/source/manual/core.rst]: How do you feel about the [logging section on the configuration page](http://www.dropwizard.io/1.0.6/docs/manual/configuration.html#logging)?

Comment (markelliot) [inline docs/source/manual/core.rst]: cool, moved there

**Comment by jplock** [comment]
@markelliot @nickbabcock should we provide a sane default and set the default to `true` (to never block)?  I'm not sure what the more explicit approach would be or what users would expect the default behavior to be.  /cc @dropwizard/committers

**Comment by markelliot** [comment]
The logback default is false, so I stuck with the existing behavior as the
default -- this seems like it will be the least surprising. Happy to flip
the default if there's a preference to do so.

**Comment by jplock** [comment]
@markelliot good call, that's probably best for now.

**Pull request merged**